juju/juju
# Add a sanitize-db.py script.

## Description of change

This generates a Javascript file that allows us to easily sanitize Juju
databases. Both their collection and their txn records are sanitized.

## QA steps

```
  $ juju bootstrap lxd
  $ juju ssh -m controller 0
  $ sudo su -
  $ service jujud-machine-0 stop
  $ ./sanitize-db.py > sanitize.js
  $ mongo CONNECT_ARGS --eval "db.txns.find().forEach(printjson)" > before.txt
  $ mongo CONNECT_ARGS ./sanitize.js
  $ mongo CONNECT_ARGS --eval "db.txns.find().forEach(printjson)" > after.txt
```

Inspect the database afterward and see that anything that could contain sensitive information like certificates or private keys has been REDACTED or removed completely. You can also `vimdiff before.txt after.txt` to see what we changed.

## Documentation changes

It should probably be referenced in documentation as something that people can use on a mongodump to allow us to introspect their database safely.

## Bug reference

Not directly. Was created to help debugging, though.
